### ISSUE TITLE
HubConnection Times out Immediately after connecting to the Hub with websocket

### ISSUE THREAD

**Issue by a6datta** [SignalR/java-client#93]
Hi I am currently working with an older version which doesn't have webSocket support. It works OK with frequent disconnection and reconnection. But the main issue I am facing is the ConcurrentModificationException #29. It crashes the Android app a few times a day. That's why I decided to downloaded the latest source code but wasn't able build the app after adding the JAR due to a java error. I then added the package directly to my app. It now builds and runs but I am getting the following errors. With the previous version Server Sent Events worked fine. It looks like when it chooses to use WebSocket that's when the problem starts. WebSocket protocol is installed in the server and my desktop application is using webSocket. I added java-websocket=1.3.1.jar to the lib too.

01-06 13:19:49.285  17604-17604/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Creating hub proxy: notificationhub
01-06 13:19:49.285  17604-17604/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubProxy NotificationHub - Subscribe to event addMessage
01-06 13:19:49.285  17604-17604/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubProxy NotificationHub - Creating new subscription for: addmessage
01-06 13:19:49.285  17604-17604/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Entered startLock in start
01-06 13:19:49.285  17604-17604/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Start the connection, using AutomaticTransport transport
01-06 13:19:49.285  17604-17604/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Start negotiation
01-06 13:19:49.285  17604-17604/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: AutomaticTransport - Start the negotiation with the server
01-06 13:19:49.295  17604-17604/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Getting connection data: [{"name":"notificationhub"}]
01-06 13:19:49.295  17604-17604/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Getting connection data: [{"name":"notificationhub"}]
01-06 13:19:49.295  17604-17604/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: AutomaticTransport - Execute the request
01-06 13:19:49.295  17604-17604/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: Create new thread for HTTP Connection
01-06 13:19:49.295  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: Execute the HTTP Request
01-06 13:19:49.295  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: URL: http://**_/signalr/signalr/negotiate?clientProtocol=1.3&connectionData=%5B%7B%22name%22%3A%22notificationhub%22%7D%5D
01-06 13:19:49.295  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: VERB: GET
01-06 13:19:49.295  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: Header User-Agent: SignalR (lang=Java; os=android; version=2.0)
01-06 13:19:49.295  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: CONTENT: null
01-06 13:19:49.295  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: Request executed
01-06 13:19:49.875  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: AutomaticTransport - Response received
01-06 13:19:49.875  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: AutomaticTransport - Read response data to the end
01-06 13:19:49.875  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: AutomaticTransport - Trigger onSuccess with negotiation data: {"Url":"/signalr/signalr","ConnectionToken":"xu8sMFVs0rQyazVovDnvswubZuFdrdiNgRECKAKXz41nhqVmxK0AwVrxae+59yk/l0Tn5sj2QJYDyxw3PYq0dL4pJX8H4noQgXh62V+uBiqSpfNJO9XmpYfRPQ894iAV","ConnectionId":"005dd70c-1ddb-45e4-83d4-5517eb768d84","KeepAliveTimeout":40.0,"DisconnectTimeout":60.0,"ConnectionTimeout":110.0,"TryWebSockets":true,"ProtocolVersion":"1.3","TransportConnectTimeout":5.0,"LongPollDelay":0.0}
01-06 13:19:49.885  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Negotiation completed
01-06 13:19:49.885  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - ConnectionId: 005dd70c-1ddb-45e4-83d4-5517eb768d84
01-06 13:19:49.885  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - ConnectionToken: xu8sMFVs0rQyazVovDnvswubZuFdrdiNgRECKAKXz41nhqVmxK0AwVrxae+59yk/l0Tn5sj2QJYDyxw3PYq0dL4pJX8H4noQgXh62V+uBiqSpfNJO9XmpYfRPQ894iAV
01-06 13:19:49.885  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Keep alive timeout: 40.0
01-06 13:19:49.885  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Entered startLock in startTransport
01-06 13:19:49.885  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Starting the transport
01-06 13:19:49.885  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Starting transport for InitialConnection
01-06 13:19:49.885  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Getting connection data: [{"name":"notificationhub"}]
01-06 13:19:49.885  17604-18743/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Getting connection data: [{"name":"notificationhub"}]
01-06 13:19:50.026  17604-18750/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Entered startLock after transport was started
01-06 13:19:50.026  17604-18750/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Current state: Connecting
01-06 13:19:50.026  17604-18750/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Starting Heartbeat monitor
01-06 13:19:50.026  17604-18750/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Connected
01-06 13:20:16.684  17604-18752/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Slow connection detected
01-06 13:20:30.028  17604-18752/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Timeout
01-06 13:20:30.028  17604-18752/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Stopping Heartbeat monitor
01-06 13:20:30.028  17604-18752/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Restarting the transport
01-06 13:20:30.028  17604-18752/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Entered startLock in startTransport
01-06 13:20:30.028  17604-18752/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Starting the transport
01-06 13:20:30.028  17604-18752/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Stopping heartbeat monitor
01-06 13:20:30.038  17604-18752/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Clearing invocation callbacks: Reconnecting
01-06 13:20:30.038  17604-18752/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Starting transport for Reconnection
01-06 13:20:30.048  17604-18752/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Getting connection data: [{"name":"notificationhub"}]
01-06 13:20:30.048  17604-18752/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Getting connection data: [{"name":"notificationhub"}]
01-06 13:20:30.399  17604-19128/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Entered startLock after transport was started
01-06 13:20:30.399  17604-19128/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Current state: Reconnecting
01-06 13:20:30.409  17604-19128/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Starting Heartbeat monitor
01-06 13:20:30.409  17604-19128/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Reconnected
01-06 13:20:30.409  17604-19128/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubProxy NotificationHub - Invoking method on hub: Client_Reconnected
01-06 13:20:30.409  17604-19128/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Registering callback: 0
01-06 13:20:30.439  17604-19128/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Sending: {"A":["**_"],"I":"0","H":"NotificationHub","M":"Client_Reconnected"}
01-06 13:20:30.439  17604-19128/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Invoking send on transport
01-06 13:20:34.843  17604-19130/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Timeout
01-06 13:20:34.843  17604-19130/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Verbose: HubConnection - Stopping Heartbeat monitor
01-06 13:20:34.843  17604-19130/com.atravesar.eiinar.eiinardriver D/SIGNALR﹕ Information: HubConnection - Restarting the transport

**Comment by a6datta** [comment]
I have just noticed even though I am sending query string 'userName' but it is always null in the server side whereas the same code works with the older version. I did have a line of code inside the reconnected event 

//mProxy.invoke("Client_Reconnected", userName).get();
after removing this line the connection becoms stable after reconnecting twice. Then the client contineouly receiving data (oing) from the server 

D/SIGNALR﹕ Verbose: HubConnection - Received data:

But before I send any messages when I check if the connection state 

if(globalVariable!=null && globalVariable.getHubConnection()!=null && globalVariable.getHubConnection().getState()== ConnectionState.Connected)
it always returns false.
